### DISCUSSION TITLE
Is destructing existing state required when actions are in a nested state?

### DISCUSSION THREAD

**Discussion by KevinMusgrave** [pmndrs/zustand#847] [Q&A] [answered]
I added an extra level to the state in [this example](https://codesandbox.io/s/new-dawn-vl03k) from [this wiki page](https://github.com/pmndrs/zustand/wiki/Splitting-the-store-into-separate-slices).

Here are my nested states:
```javascript
const createBearSlice = (set, get) => ({
  bear: {
    eatFish: () => {
      set((prev) => ({
        fish: {
          ...prev.fish,
          fishes: prev.fish.fishes > 1 ? prev.fish.fishes - 1 : 0
        }
      }));
    }
  }
});


const createFishSlice = (set, get) => ({
  fish: {
    fishes: maxFishes,
    repopulate: () => {
      set((prev) => ({ fish: { ...prev.fish, fishes: maxFishes } }));
    }
  }
});
```

To prevent the destruction of ```fish.repopulate```, I have to destructure the existing state ```...prev.fish```. I thought this isn't required because ```set``` merges state. Does it work differently when actions are nested?

Here's my code sandbox: https://codesandbox.io/s/zustand-nesting-question-bmzjxu

**Comment by dai-shi** [comment, marked as answer]
`set` merges state top-level only.
So, you need to merge it by yourself for nested states.
This is a design decision.
Reference for instance: #178
